ktor-server/ktor-server-tests/jvm/test/io/ktor/server/http/RespondWriteTest.kt

```diff
@@ -15,78 +16,71 @@ import kotlin.test.*
-                }
+    fun smoke() = testApplication {
+        routing {
+            get("/") {
+                call.respondTextWriter { write("OK") }
             }
+        }
 
-            handleRequest(HttpMethod.Get, "/").let { call ->
-                assertEquals("OK", call.response.content)
-            }
+        client.get("/").let { response ->
+            assertEquals("OK", response.bodyAsText())
         }
     }
 
     @Test
-    fun testFailureInside() {
-        withTestApplication {
-            application.routing {
-                get("/") {
-                    call.respondTextWriter {
-                        throw IllegalStateException("expected")
-                    }
+    fun testFailureInside() = testApplication {
+        routing {
+            get("/") {
+                call.respondTextWriter {
+                    throw IllegalStateException("expected")
                 }
             }
+        }
 
-            assertFailsWith<IllegalStateException> {
-                handleRequest(HttpMethod.Get, "/")
-            }
+        assertFailsWith<IllegalStateException> {
+            client.get("/")
         }
     }
 
     @Test
-    fun testSuspendInside() {
-        withTestApplication {
-            val executor = Executors.newSingleThreadExecutor()
-            application.monitor.subscribe(ApplicationStopped) { executor.shutdown() }
-            application.routing {
-                get("/") {
-                    call.respondTextWriter {
-                        withContext(executor.asCoroutineDispatcher()) {
-                            write("OK")
-                        }
+    fun testSuspendInside() = testApplication {
+        val executor = Executors.newSingleThreadExecutor()
+        application {
+            monitor.subscribe(ApplicationStopped) { executor.shutdown() }
+        }
+        routing {
+            get("/") {
+                call.respondTextWriter {
+                    withContext(executor.asCoroutineDispatcher()) {
+                        write("OK")
                     }
                 }
             }
+        }
 
-            handleRequest(HttpMethod.Get, "/").let { call ->
-                assertEquals("OK", call.response.content)
-            }
+        client.get("/").let { response ->
+            assertEquals("OK", response.bodyAsText())
         }
     }
 
-    //    @Test
-    @Suppress("UNUSED")
-    fun testFailureInsideUnresolvedCase() {
-        withTestApplication {
-            application.routing {
-                get("/") {
-                    call.respondTextWriter {
-                        close() // after that point the main pipeline is going to continue since the channel is closed
-                        // so we can't simply bypass an exception
-                        // the workaround is to hold pipeline on channel pipe until commit rather than just close
+    @Test
+    fun testFailureInsideUnresolvedCase() = testApplication {
```

Comment:
This test was disabled for some reason. Should I keep it disabled?

Reply:
Looks like it runs fine on TC, I'd say it's fine to enable.